### PR TITLE
refactor: speed up writing model fields

### PR DESCRIPTION
**Pull request by edkerk**
### Main improvements in this PR:
- refactor:
  - `importYaml` speed improvement, contributing to #387

On my PC, `importYaml('../../model/Human-GEM.yml');` tooks 248 seconds before the change and 12 seconds after the change.

**I hereby confirm that I have:**

- [x] Tested my code on my own computer for running the model
- [x] Selected `develop` as a target branch

### CODE REVIEW

**Review by mihai-sysbio — APPROVED** [review]
🦄

**Comment by edkerk** [comment]
I couldn't stand not fixing the last part. It's now much faster, takes about 12 sec. I don't see room for more improvement.

File: code/io/importYaml.m [hunk 1]
```diff
@@ -25,6 +25,25 @@
     error('Yaml file %s cannot be found', string(yamlFilename));
 end
 
+if verLessThan('matlab','9.9') %readlines introduced 2020b
+    fid=fopen(yamlFile);
```
Comment (johan-gson) [inline code/io/importYaml.m:29]: A bit confusing with yamlFile and yamlFilename - it looks like yamlFilename is the right one, if I look at line 1?

File: code/io/importYaml.m [hunk 2]
```diff
@@ -25,6 +25,25 @@
...
+    line_raw=cell(1000000,1);
+    while ~feof(fid)
+        line_raw{i}=fgetl(fid);
+        i=i+1;
+    end
+    line_raw(i:end)=[];
+    line_raw=string(line_raw);
+else
+    line_raw=readlines(yamlFile');
```
Comment (johan-gson) [inline code/io/importYaml.m:38]: should this really be yamlFile' ?

**Review by johan-gson — CHANGES_REQUESTED** [review]
Just take a look at the yamlFile things. I take it that you tested that it became faster?

File: code/io/importYaml.m (hunk 1 above)
Comment (edkerk) [inline code/io/importYaml.m:29]:
```suggestion
    fid=fopen(yamlFilename);
```

File: code/io/importYaml.m (hunk 2 above)
Comment (edkerk) [inline code/io/importYaml.m:38]:
```suggestion
    line_raw=readlines(yamlFilename');
```

Comment (edkerk) [reply code/io/importYaml.m]: Fixed, thanks for noticing!

**Review by johan-gson — APPROVED** [review]
Now it looks better!

**Comment by edkerk** [comment]
@haowang-bioinfo In f2d5352 I incorporated _similar_ changes as you introduced in #392 (but not exactly the same as it wouldn't match the refactoring that I did). Can you please check that it matches the behaviour that you need?

**Comment by haowang-bioinfo** [comment]
> @haowang-bioinfo In https://github.com/SysBioChalmers/Human-GEM/commit/f2d5352f108986ef83a0262798e35602b03db20a I incorporated similar changes as you introduced in https://github.com/SysBioChalmers/Human-GEM/pull/392 (but not exactly the same as it wouldn't match the refactoring that I did). Can you please check that it matches the behaviour that you need?

will look into this

**Comment by mihai-sysbio** [comment]
The `importYaml` function is used in the tests implemented with GH Actions. From the history I see the speedup is from [~5min30s](https://github.com/SysBioChalmers/Human-GEM/runs/6752909441?check_suite_focus=true) down to [~1m30s](https://github.com/SysBioChalmers/Human-GEM/runs/6903361764?check_suite_focus=true). With this confirmed speed improvement and tests passing, I will go ahead and merge this PR.

**Pull request merged**